cmd/submit.go

```diff
@@ -49,21 +49,69 @@ var submitCmd = &cobra.Command{
 }
 
 func runSubmit(cfg config.Config, flags *pflag.FlagSet, args []string) error {
-	usrCfg := cfg.UserViperConfig
+	if err := validateUserConfig(cfg.UserViperConfig); err != nil {
+		return err
+	}
+
+	ctx, err := newSubmitCmdContext(cfg.UserViperConfig, flags, args)
```

Comment:
This hides a lot of stuff behind it. While I think the `submitCmdContext` might be an idea worth exploring, I find it confusing to move all the logic into the `new**` function for the context.